### ISSUE TITLE
Announcement: we are looking for maintainers (you can help!)

### ISSUE THREAD

**Issue by ronkorving** [Wizcorp/phonegap-facebook-plugin#1120]
Dear community,

Over the past while you may have noticed our collective unresponsiveness to issues and pull requests. For starters, please accept our apologies. We have been rather busy and less dependent on Facebook lately, but that is no excuse to let a PhoneGap plugin as important as the Facebook one collect dust.

That's why we wanted to open the doors to add maintainers. In the spirit of what @felixge wrote about a few years ago and called [The Pull Request Hack](http://felixge.de/2013/03/11/the-pull-request-hack.html), we would like to ask who of you would be interested in being added to the list of maintainers to help get those much needed pull requests reviewed, tested and merged. We don't promise we'll accept just any application, but people who have contributed in the past or who show a sincere interest will be taken very seriously. Quite frankly, we need your help.

Applications will be reviewed by @jrouault and @aogilvie (I would like to thank @aogilvie especially for his work on this plugin so far, and I for one will very much appreciate his continued leadership here). 

Besides being a maintainer, what else can you do to help?

There are currently 219 open issues. This is way, way too much. Anyone could help out by reviewing them and helping out people with any problems they may have. Also, if any issues are false alarms, you can help by pointing out the solution. The issue tracker is like an e-mail inbox. It just never seems to shrink to zero (in fact, the opposite often seems to be true). Let's do something about this together.

We appreciate your continued support and patience.

**Comment by bau720123** [comment]
hi @ronkorving 
thanks your Announcement
I am a PHP developer but not Android and iOS
but I am glad to be a tester (help those have "false alarms" people to solve their issue)
if this plugin can use by PGB... (build.phonegap.com) (maybe publish a version to npm)

**Comment by jeduan** [comment]
Hey @ronkorving @aogilvie it's great asking for help. I have a popular fork of this plugin that fixes many popular issues such as support for npm, Cordova 4, Facebook SDK 4, iOS 9 and App Invites. I'd love to see it merged upstream. I'd like to start a discussion in that regard as many issues could be completely fixed out of the bat.

**Comment by ronkorving** [comment]
@bau720123 that's fantastic, much appreciated!

**Comment by jrouault** [comment]
Hello @jeduan! First off, thank you for taking the time to answer and for contributing to this plugin through your fork.
We have added you as a maintainer (let me know if I assumed wrong from your message) and you are free to review, test and merge contributions (hopefully in this order :smiley:) from fellow developers.

It would be nice indeed to have a discussion about the work you have done, as @mereformalities has done a similar job to support the Facebook SDK 4.1.
We do not mind merging your work upstream. It would be nice to have all these issues fixed as soon as possible. However we would like it to be in respect to the contribution process (described in [CONTRIBUTING.md](https://github.com/Wizcorp/phonegap-facebook-plugin/blob/master/CONTRIBUTING.md)) even as maintainer.
Feel free to open a PR to start the discussion, allow people to review it and test it before merging it.

If you have any questions, please do ask. We will happily answer them.

**Comment by jrouault** [comment]
@bau720123 Thanks for your reply!
Thank you for willing to help out as a tester. If you wish to do so you can already have a look at the list of Pull Requests and comment on it to give a feedback. It would be really great and will help to decide whether a PR should be merged or not.
Concerning npm, as @ronkorving replied, this is quite an easy task which will be done asap.

**Comment by aogilvie** [comment]
The more testers the better. Testing PhoneGap plugins on all devices and various OS versions is quite time consuming. Sharing these QA tasks among the community would greatly help.

FYI, Cordova 4 & FB SDK 4 support is in dev branch though we need lots of testing before rolling it out to master branch.

A lot of the old issues & PRs will be immediately close-able because the FB SDK 4 removes a lot of the bugs previously found.

Regarding npm etc. myself or the Wizcorp team can add users to our npm account etc. for publishing new versions.

**Comment by bau720123** [comment]
hi all
I recommend we should made two facebook plugin，one is the stable version，another is the tester version (I don't know if it is make sense，just recommanded，if wrong，forgive me)
(PGB(build.phonegap.com) environment is my most familiar，phonegap or cordova cli is the next....)

for example︰
cordova-plugin-stablefacebook︰after the testing，all success "merge" will publish to this version
cordova-plugin-betafacebook︰all merge will all under here，and I(or somebody) will testing these one by one，test success merge will Classify to "cordova-plugin-stablefacebook"，others will Classify to "testing fail"

**Comment by ronkorving** [comment]
@bau720123 can you make an issue about this topic, so we can discuss it there?

**Comment by bau720123** [comment]
hi @ronkorving 
I don't know what you mean
make an issue about this topic ???
or you want me to post a "new issue" that about two plugin thing (one is stable the other is beta) ?

**Comment by aogilvie** [comment]
But getting back to scope, I'd like to specifically ask (beg?) some previous contributors to claim maintainer rights: @keab42 @sgrebnov @EddyVerbruggen @dnoliver @ripcurlx

**Comment by ripcurlx** [comment]
Hi Ally!

I can't promise that I'm able to contribute lots of my time, but I'll try my best to help out here and there.
So feel free to add me as a maintainer :-)

Best,

Christoph

**Comment by ronkorving** [comment]
@bau720123 yes please, create a new issue about how to deal with beta and stable (so that we can keep this issue thread on-topic).

**Comment by bau720123** [comment]
hi @ronkorving 
I have already post a new one under here
https://github.com/Wizcorp/phonegap-facebook-plugin/issues/1121

the original have been delete

**Comment by dnoliver** [comment]
I can contribute with jasmine tests (automatic and manual tests) :) . That is what I do the best!

**Comment by sgrebnov** [comment]
It will be my pleasure to help maintaining this plugin moving forward...

**Comment by matiassingers** [comment]
I have this repo on watch, and will help as much as I can. I'm not actively developing Cordova apps at this point, but can still help triaging issues etc.

**Comment by jrouault** [comment]
Thanks @ripcurlx @dnoliver @sgrebnov @matiassingers for answering the call! You have all been added as maintainers.

**Comment by sgrebnov** [comment]
:rocket: Awesome!

**Comment by keab42** [comment]
I'm hoping to have more time to be able to contribute to this again. Please feel free to add me.

**Comment by jrouault** [comment]
Thanks @keab42 for willing to contribute to this plugin! You have been added as a maintainer too.

**Comment by nikhilkh** [comment]
@jeduan are the changes from your fork been merged? What's the release plan for the plugin to npm?

**Comment by jeduan** [comment]
@nikhilkh They haven't. I had to take a ~6 weeks absence from open source so I haven't been able to organize everything in a way that can be sent as a pull-request.

**Comment by nikhilkh** [comment]
Fair enough - could you summarize - what are the key fixes that your fork includes?

**Comment by quedicesebas** [comment]
Go @jeduan !!! we need facebook SDK 4
